mcm/tests/api_tools.py

```diff
@@ -0,0 +1,496 @@
+                ]
+
+        # Check and raise
+        if error_msg:
+            compress_msg = "\n".join(error_msg)
+            raise RuntimeError(compress_msg)
+
+        # Check the required mock users are available
+        # Otherwise, create them
+        if not mockup:
+            if not self.check_test_users():
+                self.__include_test_users()
+
+    def __http_request(
+        self,
+        base: str,
+        endpoint: str,
+        method: str,
+        data: dict = {},
+        headers: dict = {},
+    ) -> Response:
+        """
+        Executes a HTTP request.
+
+        Args:
+            base (str): Base URL, e.g: http://localhost:5984/
+            endpoint (str): Endpoint to consume, e.g: entity/document?p=1
+            method (str): HTTP method, e.g: POST
+            data (dict | None): Data to include in the request body.
+            headers (dict): Request headers
+        """
+        session = Session()
+        full_url: str = base + endpoint
+        req = Request(method=method, url=full_url, headers=headers, json=data)
+        prepared_req = session.prepare_request(req)
+        return session.send(prepared_req)
+
+    def couchdb_request(
+        self, endpoint: str, method: str, data: dict = {}, headers: dict = {}
+    ) -> Response:
+        """
+        Sends a HTTP request to CouchDB.
+
+        Args:
+            endpoint (str): Endpoint to consume, e.g: entity/document?p=1
+            method (str): HTTP method, e.g: POST
+            data (dict | None): Data to include in the request body.
+            headers (dict): Request headers
+        """
+        couchdb_headers = {"Authorization": self.mcm_couchdb_credential}
+        full_headers = {**headers, **couchdb_headers}
+        return self.__http_request(
+            base=self.mcm_couchdb_url,
+            endpoint=endpoint,
+            method=method,
+            data=data,
+            headers=full_headers,
+        )
+
+    def mcm_request(
+        self,
+        endpoint: str,
+        method: str,
+        as_role: Roles | None = None,
+        data: dict = {},
+        headers: dict = {},
+    ) -> Response:
+        """
+        Sends a HTTP request to McM web application.
+
+        Args:
+            endpoint (str): Endpoint to consume, e.g: entity/document?p=1
+            method (str): HTTP method, e.g: POST
+            as_role (Roles | None): Role to impersonate for the request.
+            data (dict | None): Data to include in the request body.
+            headers (dict): Request headers
+        """
+        full_headers = {**headers}
+        if as_role:
+            user_headers = {"Adfs-Login": as_role.value}
+            full_headers.update(user_headers)
+
+        return self.__http_request(
+            base=self.mcm_application_url,
+            endpoint=endpoint,
+            method=method,
+            data=data,
+            headers=full_headers,
+        )
+
+    def lucene_request(
```

Comment:
Provide a function to create Lucene URLs instead.

Reply:
Also refactored